File: TripKitAndroidUIModules/TripKitAndroidUIData/src/main/java/com/skedgo/tripkit/ui/data/tripprogress/LocationSampleDto.java
package com.skedgo.tripkit.ui.data.tripprogress;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson.TypeAdapters;
import org.immutables.value.Value.Immutable;

import static org.immutables.value.Value.Style;
import static org.immutables.value.Value.Style.ImplementationVisibility.PACKAGE;

@Immutable
@Style(
    overshadowImplementation = true,
    passAnnotations = JsonAdapter.class
)
@TypeAdapters
@JsonAdapter(GsonAdaptersLocationSampleDto.class)
public interface LocationSampleDto {
  long timestamp();
  double latitude();
  double longitude();
  int bearing();
  int speed();

  class Builder extends ImmutableLocationSampleDto.Builder {}
}

File: TripKitAndroidUIModules/TripKitAndroidUIData/src/main/java/com/skedgo/tripkit/ui/data/tripprogress/LocationSampleDto.kt
package com.skedgo.tripkit.ui.data.tripprogress
import com.skedgo.tripkit.location.LocationSample

internal fun LocationSample.toLocationSampleDto(): LocationSampleDto =
    LocationSampleDto.Builder().apply {
      timestamp(timestamp)
      latitude(latitude)
      longitude(longitude)
      bearing(bearing)
      speed(speed)
    }.build()